### DISCUSSION TITLE
v1.17

### DISCUSSION THREAD

**Discussion by duracelltomi** [duracelltomi/gtm4wp#269] [General]
No change since the beta, repeating changelog for easier access:

* Added: pagePostTerms data layer variable will now also include a meta key with post meta values that does not start with the _ character. This should allow to utilize custom fields added by plugins like Advanced Custom Fields.
* Added: new filter: gtm4wp_post_meta_in_datalayer to limit which post meta should be present with with the previously added feature.
* Added: ability to use a custom path for your server side GTM container. This makes GTM4WP compatible with services like stape.io
* Added: new advanced option to disable browser console.log() messages on frontend
* Added: new filter: gtm4wp_purchase_datalayer to be able to alter data layer content for the purchase data
* Added: new filter: gtm4wp_datalayer_on_pageload to be able to alter data layer content generated during page load
* Added: customerBillingState and customerShippingState on WooCommerce order received page, thanks [massimo-maimeri](https://github.com/massimo-maimeri)
* Added: High Performance Order Storage (HPOS) compatibility for WooCommerce
* Updated: WooCommerce integration will also check target attitbute of product links while firing select_item event, thanks [robklo])(https://github.com/robklo)
* Updated: YouTube tracking will not load on pages not containing a YouTube block or a YouTube iframe
* Fixed: add_to_cart event on grouped products not working when SKU is selected as product ID in plugin options.
* Fixed: more reliable new_customer flag working with guest orders of existing customers, thanks [morvy](https://github.com/morvy)

<hr /><em>This discussion was created from the release <a href='https://github.com/duracelltomi/gtm4wp/releases/tag/1.17'>v1.17</a>.</em>

**Comment by rashed0101** [comment]
Purchase datalayer is not showing on Custom thank you. Other tags are working properly.

**Reply by duracelltomi** [reply]
That depends on the 3rd party plugin that generates the custom thank you page.
GTM4WP queries WooCommerce whether the user is on the order received page. A 3rd party plugin can override this query and tell everyone that the custom page IS the order received page. If it is not done, GTM4WP will not output the purchase data layer.

**Reply by rashed0101** [reply]
Could you please help me, how can i do that? which plugin should i use? How can i solve this problem?

**Reply by duracelltomi** [reply]
As far as I know, this plugin worked well, but I did not test it in the past 1-2 years. Earlier it worked well:
https://wordpress.org/plugins/woo-thank-you-page-nextmove-lite/
